pytest/tests/sanity/upgradable.py

```diff
@@ -170,167 +147,119 @@ def send_transfer(self) -> None:
             self._rpc_node.get_account(account_id)['result']['amount'])
         assert hex_account_balance == amount
 
-    def assert_no_failed_txs(self) -> None:
-        assert self._failed_txs == 0, f"{self._failed_txs} transactions failed"
-
 
-@dataclasses.dataclass
 class Protocols:
-    stable: int
-    current: int
 
-    @classmethod
-    def from_executables(
-        cls,
-        executables: branches.ABExecutables,
-    ) -> 'Protocols':
-        _, stable = get_proto_version(executables.stable.neard)
-        _, current = get_proto_version(executables.current.neard)
-        assert current >= stable, "cannot downgrade protocol version"
-        return cls(stable, current)
+    def __init__(self, executables: branches.ABExecutables):
+        self.stable = get_proto_version(executables.stable.neard)
+        self.current = get_proto_version(executables.current.neard)
+        assert self.current >= self.stable, "cannot downgrade protocol version"
 
 
 class TestUpgrade:
 
-    def __init__(
-        self,
-        num_validators: int,
-        node_prefix: str,
-        epoch_length: int,
-    ) -> None:
-        self._num_validators = num_validators
-        self._node_prefix = node_prefix
-        self._epoch_length = epoch_length
-
-        self._executables = get_executables()
-        self._protocols = Protocols.from_executables(self._executables)
+    def __init__(self) -> None:
+        self._executables = branches.prepare_ab_test()
+        self._protocols = Protocols(self._executables)
         node_dirs = self.configure_nodes()
         nodes = self.start_nodes(node_dirs)
+        time.sleep(5)  # Give some time for nodes to start
```

Comment:
it's better practice to wait until blocks start appearing, it's much faster and safer than just a timeout